lib/api/api.js

```diff
@@ -67,6 +72,17 @@ const api = {
             // Pull the source bucket and source object separated by /
             sourceBucket = source.slice(0, slashSeparator);
             sourceObject = source.slice(slashSeparator + 1);
+            sourceVersionId = query ?
+                querystring.parse(query).versionId : undefined;
+            sourceVersionId = sourceVersionId || undefined;
+            if (sourceVersionId && sourceVersionId !== 'null') {
+                try {
+                    sourceVersionId = VID.decrypt(sourceVersionId);
+                } catch (exception) {
+                    return callback(errors.InvalidArgument
+                        .customizeDescription('Invalid version id specified'));
+                }
+            }
```

Comment:
That's tricky. Do you mean you want to send the copy source version id in the regular (`x-amz-version-id`) place in the headers? Something like this in lib/api/apiUtils/authorization/prepareRequestContexts.js:

```
export default function prepareRequestContexts(apiMethod, request, sourceBucket,
    sourceObject, sourceVersionId) {
[...]
    if (apiMethod === 'objectCopy' || apiMethod === 'objectPutCopyPart') {
       const requestHeaders = Object.assign{{}, request.headers, { 'x-amz-version-id': sourceVersionId });
        const getRequestContext = new RequestContext(request.headers,
            request.query, sourceBucket, sourceObject,
            request.socket.remoteAddress, request.connection.encrypted,
            'objectGet', 's3');
[...]
```